search/query/knn.go

```diff
@@ -35,7 +35,10 @@ type KNNQuery struct {
 	BoostVal    *Boost    `json:"boost,omitempty"`
 
 	// see KNNRequest.Params for description
-	Params json.RawMessage `json:"params"`
+	Params           json.RawMessage `json:"params"`
+	FilterQuery      Query           `json:"filter,omitempty"`
+	FilterResults    []index.IndexInternalID
+	RequireFiltering bool
```

Comment:
Don't quite understand the need for this.

Reply:
@abhinavdangeti this is used downstream to indicate whether a KNN query requires filtering(is a filtered query) or not(does not have a filter query).

Reply:
Removed it in my recent commits, since I've made some changes to the overall approach.